rules/terraformrules/terraform_naming_convention.go

```diff
@@ -0,0 +1,292 @@
+package terraformrules
+
+import (
+	"fmt"
+	"log"
+	"regexp"
+
+	"github.com/terraform-linters/tflint/tflint"
+)
+
+// TerraformNamingConventionRule checks whether blocks follow naming convention
+type TerraformNamingConventionRule struct{}
+
+type terraformNamingConventionRuleConfig struct {
+	BlockFormatConfig
```

Comment:
`format` and `custom` tags are not recognized when embedding a struct. It needs to be modified as follows:

```suggestion
	Format string `hcl:"format,optional"`
	Custom string `hcl:"custom,optional"`
```

By the way, embedding of `Data` etc. works correctly because the  `hcl:"data,block"` tag is recognized to the struct itself.